flyerhq/flutter_chat_ui
# feat: add unseen messages banner and scroll to first unseen functionality

### What does it do?

Add an unread messages banner and the option to scroll to the first unread message on chat open.

<img width="430" alt="Screenshot 2022-01-31 at 14 42 19" src="https://user-images.githubusercontent.com/27334472/151804187-1768566c-57de-4a5e-baf1-92b90cf28d0a.png">

### Why is it needed?

Users should have a way of being brought directly to the content they need to see most urgently. In many cases this is the first unread message.

### How to test it?

In the example app, I have set `scrollToUnseenOnOpen` so it should immediately scroll to the first message.

### Related issues/PRs

Closes #179 and closes #213

This is built on top of #208 and should not be merged before it. Second part of #179.

### Open Questions / TODO

- [x] Should there be an option to disable the banner too?
- [x] **Designer:** I need designs for the banner and then I can also make them configurable in the ChatTheme
- [x] I will write some tests for my changes to the 'util' function
- [x] The unseen message calculation currently only works if the status is set correctly for each message. I wonder whether the statuses for received messages are even really relevant and can be used like this. Wouldn't it make more sense to have a `seenBy` array on messages to check for the first message which is not marked as seen by the current user?
- [x] Scroll to bottom indicator (separate issue, tackle later)
